uaclient/api/u/pro/subscription/v1.py

```diff
@@ -0,0 +1,265 @@
+            doc="List of products included in the contract",
+        ),
+        Field(
+            "tech_support_level",
+            StringDataValue,
+            doc="Technical support level of the contract",
+        ),
+        Field(
+            "origin",
+            StringDataValue,
+            required=False,
+            doc="Origin of the contract",
+        ),
+        Field(
+            "effective",
+            DatetimeDataValue,
+            required=False,
+            doc="Effective start date of the contract",
+        ),
+        Field(
+            "expires",
+            DatetimeDataValue,
+            required=False,
+            doc="Expiration date of the contract",
+        ),
+    ]
+
+    def __init__(
+        self,
+        *,
+        created_at: str,
+        id: str,
+        name: str,
+        products: List[str],
+        tech_support_level: str,
+        origin: Optional[str],
+        effective: Optional[str],
+        expires: Optional[str],
+    ):
+        self.created_at = created_at
+        self.id = id
+        self.name = name
+        self.products = products
+        self.tech_support_level = tech_support_level
+        self.origin = origin
+        self.effective = effective
+        self.expires = expires
+
+
+class AccountInfo(DataObject, AdditionalInfo):
+    fields = [
+        Field(
+            "created_at", DatetimeDataValue, doc="Creation date of the account"
+        ),
+        Field(
+            "external_account_ids",
+            data_list(StringDataValue),
+            doc="External account identifiers",
+        ),
+        Field("id", StringDataValue, doc="Account id"),
+        Field("name", StringDataValue, doc="Account name"),
+    ]
+
+    def __init__(
+        self,
+        *,
+        created_at: str,
+        external_account_ids: List[str],
+        id: str,
+        name: str,
+    ):
+        self.created_at = created_at
+        self.external_account_ids = external_account_ids
+        self.id = id
+        self.name = name
+
+
+class SubscriptionResult(DataObject, AdditionalInfo):
+    fields = [
+        Field("contract", ContractInfo, doc="Contract Information"),
+        Field("account", AccountInfo, doc="Account information"),
+        Field("machine_id", StringDataValue, doc="Machine id"),
+        Field("activity_id", StringDataValue, doc="Activity id"),
+    ]
+
+    def __init__(
+        self,
+        *,
+        contract: ContractInfo,
+        account: AccountInfo,
+        machine_id: str,
+        activity_id: str,
+    ):
+        self.contract = contract
+        self.account = account
+        self.machine_id = machine_id
+        self.activity_id = activity_id
+
+
+def subscription() -> SubscriptionResult:
+    return _subscription(UAConfig())
+
+
+def _subscription(cfg: UAConfig) -> SubscriptionResult:
+    """
+    Returns the Ubuntu Pro subscription information for the machine.
+    """
+    if not _is_attached(cfg).is_attached:
```

Comment:
should we have a `is_attached` field here ? Just to easily allow users to know that they are running on an unattached machine